### DISCUSSION TITLE
How do we use svg icons with this plugin?

### DISCUSSION THREAD

**Discussion by mybardaklar** [webdiscus/pug-plugin#45] [Q&A]
Hi, its an amazing plugin!

But i have a question. How can i use svg icons with this plugin? There is no information in documents about this. For example i was using posthtml-inline-svg with posthtml-loader. But with this plugin i cannot use posthtml-loader. With posthtml-inline-svg, i can use some svg attrs like witdh, height, fill and other attrs. But how am i gonna use kind of things with this plugin ?

Thank you.

**Comment by webdiscus** [comment]
Hi @mybardaklar,

you can see the usage examples on demo site [hello-world](https://webdiscus.github.io/pug-plugin/hello-world/) and their [sources](https://github.com/webdiscus/pug-plugin/tree/master/examples/hello-world).

Pug plugin add all SVG-image attributes to inline SVG-tag:
```pug
img.logo(src=require('logo-image.svg') with=100 height=200)
```
Generated HTML
```html
<svg viewBox="0 0 200 200" class="logo" with=100 height=200>
  <path d="m600 0 530.3 300v600l-530.3 300-530.3-300v-600z" fill="#fff"></path>
  ...
</svg>
```

Pug plugin can inline SVG/PNG/JPG/etc using Webpack option `parser.dataUrlCondition.maxSize` for `type: 'asset'` or using a query, e.g. `image.svg?inline`, in this case must be added follow rule:
```js
   // force inline images containing `?inline` query
    {
      test: /\.(png|svg)$/i,
      type: 'asset/inline',
      resourceQuery: /inline/,
    },
```
See please other usage examples [webpack-pug-scss-js-starter](https://github.com/webdiscus/webpack-pug-scss-js-starter)

> **Note**
>
> Pug plugin not require additional plugins for inline SVG.

**Reply by mybardaklar** [reply]
Yeah, thank you for this. But there is also problem like this. My font folder structure is like this "src/fonts/FontName/...font files". But when compile comleted, dist is like this for fonts "fonts/...font files". But i want it like this "dist/fonts/FontName/...font files". Because i can have more than one folders and files. Thank you.

**Reply by webdiscus** [reply]
Yet one important thing - if you have both fonts and images as SVG then use `include` option by a rule.
To keep folder name of fonts in output directory use the option `filename` as function, see example below.

```js
        {
          test: /\.(png|jpg|jpeg|svg|ico)$/,
          type: 'asset/resource',
          include: /\/images/, // <= use RegExp for folder with SVG images
          generator: {
            filename: 'assets/img/[name].[hash:8][ext][query]',
          },
        },

        {
          test: /\.(woff|woff2|svg|eot|ttf|otf)$/i,
          type: 'asset/resource',
          include: /\/fonts/,  // <= use RegExp for folder with SVG fonts
          generator: {
            // generate filename include last dirname to group fonts by name
            filename: (pathData) => `assets/fonts/${path.basename(path.dirname(pathData.filename))}/[name][ext][query]`,
          }, // => dist/assets/fonts/FontName/font-file.svg
        },
```

**Reply by mybardaklar** [reply]
Thank you so much. I already used RegExp for include rule. But what if i have more subfolders? That is a problem i think.

**Reply by webdiscus** [reply]
for example?

**Reply by mybardaklar** [reply]
"src/fonts/ParentFolder/SubFolder/...fontfiles" like this.

**Reply by webdiscus** [reply]
you can self write a logic in `filename` function for your folder structure.
See please Webpack docu for [filename](https://webpack.js.org/configuration/output/#outputfilename).
This case has nothing to do with Pug plugin ;-)

**Reply by mybardaklar** [reply]
Yes, you are right. Sorry. Thank you for these informations.
